src/feature.c

```diff
@@ -0,0 +1,141 @@
+/* Copyright (C) 2007-2010 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Jeff Lucovsky <jeff@lucovsky.org>
+ *
+ * Implements feature tracking
+ */
+
+#include "suricata-common.h"
+#include "feature.h"
+#include "debug.h"
+
+#include "util-hash.h"
+
+typedef struct _FeatureEntryType {
```

Comment:
I don't remember why I started with the trailing underscore in this case, but I do remember why it is trailing and not leading. See for example https://doxygen.openinfosecfoundation.org/annotated.html The sort order is alphabetical and the leading underscores affect this.

But recently I started leaving the underscore out completely, so I think we can just drop it. If there is a reason to add it, please make it trailing.

Reply:
Will address.